uber/custom_tags.py

```diff
@@ -1,74 +1,136 @@
+"""
+Portions of this file are copied from the Django project. Relevant portions are
+Copyright (c) Django Software Foundation and individual contributors.
+Licensed under BSD license, see full license for details:
+https://github.com/django/django/blob/4696078832f486ba63f0783a0795294b3d80d862/LICENSE
+"""
+
 from uber.common import *
 
 
-@register.filter
-def datetime(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
-    return ' '.join(dt.astimezone(c.EVENT_TIMEZONE).strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm')
+def safe_string(str):
+    return jinja2.Markup(str)
+
 
-from datetime import datetime  # noqa: now that we've registered our filter, re-import the "datetime" class to avoid conflicts
+@JinjaEnv.jinja_filter(name='date')
+def date_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
+    # TODO: if this is a datetime.time object, do we have to correct for timezone with .astimezone()?
+    return ' '.join(dt.strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm') if dt else ''
```

Comment:
Regarding datetime formatting filters and timezones, I'd like to change all of our filters to use the following convention: if the filter has the `_local` suffix, it converts the datetime argument to the event's timezone; if the filter has no suffix, it format's the datetime without timezone conversion.

```
@JinjaEnv.jinja_filter(name='datetime')
def datetime_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
    return ' '.join(dt.strftime(fmt).split()).replace('AM', 'am').replace('PM', 'pm') if dt else ''


@JinjaEnv.jinja_filter(name='datetime_local')
def datetime_local_filter(dt, fmt='%-I:%M%p %Z on %A, %b %e'):
    return datetime_filter(dt.astimezone(c.EVENT_TIMEZONE), fmt=fmt)
```

We already use this convention for our model datetime attributes. By adopting this convention for our filters as well, a naive developer will be able to read our templates and always know if the datetime is in UTC or the event's local time.

Thoughts?